src/sync/impl/apple/network_reachability_observer.hpp

```diff
@@ -0,0 +1,62 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_NETWORK_REACHABILITY_OBSERVER_HPP
+#define REALM_OS_NETWORK_REACHABILITY_OBSERVER_HPP
+
+#include <functional>
+#include <string>
+
+#include <realm/util/cf_ptr.hpp>
+#include <realm/util/optional.hpp>
+
+#include <SystemConfiguration/SystemConfiguration.h>
+
+namespace realm {
+
+enum NetworkReachabilityStatus {
+    NotReachable,
+    ReachableViaWiFi,
+    ReachableViaWWAN
+};
+
+class NetworkReachabilityObserver {
```

Comment:
Talking to @stel and @fealebenpae yesterday about this. Both Java and .NET feels this is to complicated for what we need. Just having the `SyncManager::reconnect()` or similar would be sufficient for us since all code to detect network changes would be in the binding anyway. 

Having an observer pattern on the SyncManager is also slightly weird, since it is the SyncManager interested in Network changes, not the other way around.